ocw/lib/azure.py

```diff
@@ -141,32 +160,45 @@ def parse_image_name(self, img_name):
         ]
         return self.parse_image_name_helper(img_name, regexes)
 
-    def cleanup_sle_images_container(self, bbsrv, container_name):
-        generator = bbsrv.list_blobs(container_name)
-        images = dict()
-        for img in generator:
+    def cleanup_sle_images_container(self, bbsrv, keep_images):
+        container_name = 'sle-images'
+        for img in bbsrv.list_blobs(container_name):
             m = self.parse_image_name(img.name)
-            if (m):
-                key = m['key']
-                if key not in images:
-                    images[key] = list()
-
-                logger.debug('[{}]Image {} is candidate for deletion with build {} '.format(
+            if m:
```

Comment:
but you already pass this collection through `parse_image_name` , so why we need to do it second time ?

Reply:
ah wait , that was different collection ... hm we need to discuss this :)

Reply:
This avoid of deleting arbitrary other disc/images which might be in the rg as well.